File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
  
  if (nums.length === 1) {
    return 0;
  }

  var sumLeft = [];
  var sumRight = [];

  //iterate through nums and update sumleft and sumright, compare values
  for (var i = 0; i < nums.length; i++) {
    
    var leftSum = sumLeft.reduce(function (acc, item) {
        return acc + item;
      }, 0)
    
    sumRight = nums.slice(i+1, nums.length);
    
    var rightSum = sumRight.reduce(function (acc, item) {
      return acc + item;
    }, 0)
    
  //if they're ever equal return i
    if (rightSum === leftSum) {
      return i;
    } 
    
    sumLeft.push(nums[i]);
  }
  
  return -1
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        '''

        calculate total sum
        leftSum + nums[i] + rightSum will always equal sum(nums)

        maintain leftmost pivot index
        there can be multiple pivot indices
        '''
        
        leftSum = 0
        rightSum = sum(nums)
        for i in range(0, len(nums)):
          # update rightSum
          rightSum -= nums[i]

          # compare, update res
          if leftSum == rightSum:
            return i

          # update leftSum
          leftSum += nums[i]

        return -1

